doc/conf.py

```diff
@@ -96,6 +103,15 @@
 
 autodoc_docstring_signature = True
 
+
+# Sphinx gallery configuration
+sphinx_gallery_conf = {
+   # path to your examples scripts
+    'examples_dirs' : '../examples',
+    # path where to save gallery generated examples
+    'gallery_dirs'  : 'auto_examples'}
```

Comment:
It is possible, but (1) so far there has been no compelling reason to do it; (2) the amount of work required to change this is sufficiently large that I am not doing it.

To extend on point 1, our naming patterns for example plots and links is obscure, inconsistent and so poorly descriptive of what is in the examples that I really don't think the name of the folder is ever going to help or prevent our users from finding anything (we have about 5 examples named "demo bar_X", with X an incremental number). We also need to be realistic: no one is ever going to guess our url or even attempt to guess our url, and possibly not even look at the url.
Second, it is a convention that is followed by most projects and following conventions have many benefits for easing contribution, maintenance, etc. 
Third, taking personal preferences into account is not for me a compelling reason (I personally prefer auto_examples over both examples and mpl_examples, as it makes it to a certain extent more clear that this folder is auto-generated and thus should not be modified by our contributors, and this is typically the kind of questions where if you have N developers in the room, you'll have N opinions).